packages/protocol/src/action-protocol/client-notification.ts

```diff
@@ -114,3 +108,130 @@ export namespace ServerMessageAction {
+
+    /**
+     * An ID that can be used in subsequent `updateProgress` and `endProgress` events to make them refer to the same progress reporting.
+     */
+    progressId: string;
+    /**
+     * Short title of the progress reporting. Shown in the UI to describe the long running process.
+     */
+    title: string;
+    /**
+     * Optional additional progress message. Shown in the UI to describe the long running process.
+     */
+    message?: string;
+    /**
+     * Progress percentage to display (value range: 0 to 100). If omitted no percentage is shown.
+     */
+    percentage?: number;
+}
+
+export namespace StartProgressAction {
+    export const KIND = 'startProgress';
+
+    export function is(object: any): object is StartProgressAction {
+        return Action.hasKind(object, KIND) && hasStringProp(object, 'progressId') && hasStringProp(object, 'title');
+    }
+
+    export function create(
+        progressId: string,
+        title: string,
+        options: {
+            message?: string;
+            percentage?: number;
+        } = {}
+    ): StartProgressAction {
+        return {
+            kind: KIND,
+            progressId,
+            title,
+            ...options
+        };
+    }
+}
+
+/**
+ * Sent by the server to the client to presenting an update of the progress of a long running process in the UI.
+ */
+export interface UpdateProgressAction extends Action {
+    kind: typeof UpdateProgressAction.KIND;
+
+    /**
+     * The ID of the progress reporting to update.
+     */
+    progressId: string;
+    /**
+     * The message to show in the progress reporting.
+     */
+    message?: string;
+    /**
+     * The percentage (value range: 0 to 100) to show in the progress reporting.
+     */
+    percentage?: number;
+}
+
+export namespace UpdateProgressAction {
+    export const KIND = 'updateProgress';
+
+    export function is(object: any): object is UpdateProgressAction {
+        return Action.hasKind(object, KIND) && hasStringProp(object, 'progressId');
+    }
+
+    export function create(
+        progressId: string,
+        options: {
+            message?: string;
+            percentage?: number;
+        } = {}
+    ): UpdateProgressAction {
+        return {
+            kind: KIND,
+            progressId,
+            ...options
+        };
+    }
+}
+
+/**
+ * Sent by the server to the client to end the reporting of a progress.
+ */
+export interface EndProgressAction extends Action {
+    kind: typeof EndProgressAction.KIND;
+
+    /**
+     * The ID of the progress reporting to update.
+     */
+    progressId: string;
+    /**
+     * The message to show in the progress reporting.
+     */
+    message?: string;
+}
+
+export namespace EndProgressAction {
+    export const KIND = 'endProgress';
+
+    export function is(object: any): object is EndProgressAction {
+        return Action.hasKind(object, KIND) && hasStringProp(object, 'progressId');
+    }
+
+    export function create(
```

Comment:
I don't know if we have other occurrences of this pattern but if the create function has one required and one optional parameter wrapping the optional parameter in an options object seems a bit overkill to me.